src/lib/support/Span.h

```diff
@@ -42,6 +45,17 @@ class Span
     constexpr explicit Span(T (&databuf)[N]) : Span(databuf, N)
     {}
 
+    // Allow implicit construction from a Span over a type that matches our
+    // type, up to const-ness.
+    template <class U, typename = std::enable_if_t<std::is_same<std::remove_const_t<T>, std::remove_const_t<U>>::value>>
+    constexpr inline Span(const Span<U> & other) : Span(other.data(), other.size())
```

Comment:
nit: `inline` does nothing here. Both because it's defined in the class + `constexpr` implies `inline`.

Reply:
Done.